staging/kos/pkg/controllers/subnet/validator.go

```diff
@@ -156,13 +155,30 @@ func (v *Validator) OnUpdate(oldObj, newObj interface{}) {
 	oldS, newS := oldObj.(*netv1a1.Subnet), newObj.(*netv1a1.Subnet)
 	glog.V(5).Infof("Notified of update from %#+v to %#+v.", oldS, newS)
 
-	// Process a subnet only if the fields that affect validation have changed.
-	if oldS.Spec.IPv4 != newS.Spec.IPv4 || oldS.Spec.VNI != newS.Spec.VNI {
-		v.queue.Add(k8stypes.NamespacedName{
-			Namespace: newS.Namespace,
-			Name:      newS.Name,
-		})
+	nsn := k8stypes.NamespacedName{
+		Namespace: newS.Namespace,
+		Name:      newS.Name,
 	}
+
+	// Since only VNI and CIDR block affect validation and they're immutable
+	// don't enqueue on update notifications, except for those which are caused
+	// by a delete followed by a re-create (UID change) or those caused by
+	// status updates from this controller so that all the processing which was
+	// not performed because the subnet in the Informer's cache was stale can be
+	// performed.
+	if oldS.UID != newS.UID || v.subnetIsAwaited(nsn, newS.ResourceVersion) {
+		v.queue.Add(nsn)
+	}
+}
+
+func (v *Validator) subnetIsAwaited(subnet k8stypes.NamespacedName, rv string) bool {
```

Comment:
The name "subnetIsAwaited" has positive connotations, whereas the actual condition here is more of a negative.  I suggest a different name, perhaps something like "subnetStaleNoMore", and/or adding an explanatory comment.

Reply:
This can be handled in a follow-up PR.